### PR TITLE
fix: preview not working during festival

### PR DESCRIPTION
**Pull request by AlejandroAlvarezMelucciDCL**
- Refactor on how we set the start position with possible overrides
- Removes the logic from `RealmNavigator`
- Takes into consideration a position set by args so the feature flag is ignore accordingly
- Also sets the correct value for `localhost` in `Hostname` for feature flag usage.

Fixes: https://github.com/decentraland/creator-hub/issues/312

### QA TEST STEPS

- Start the client with the flags `--debug --dclenv zone`. You should start in 100,100
- Start the client with the flags `--debug --dclenv zone --position 150,150`. You should start in 150,150
- Start the client with the flags `--debug --dclenv zone --realm olavra.dcl.eth`. Dont mind that the scene doesnt load, just check that the coordinates are `0,0`. (Also, move a little bit through the scene, probably the scene will say `-150,-150`. If you move a parcel it should change correclty)

Local scene

1. Download [this test scene](https://github.com/user-attachments/files/17851248/quick-00-scene-test.zip) and uncompress it somewhere
2. Enter the scene root folder with a terminal/command and run `npm i` and then `npm run start`
3. Close the Explorer that auto-opened. Leave the scene running in the console/terminal.
4. Download the build from this PR and connect it to the running scene using [the console command](https://github.com/decentraland/unity-explorer/wiki/How-to-connect-to-a-local-scene-(Unity-Editor---Custom-Build---Latest-Released-Build)#connecting-a-custom-build-to-the-scene) according to your OS. Add the `--dclenv zone` as well to the args
5. The client should open and the scene with the cube in the middle should be shown
6. Close the client
7. Close the terminal/command running the scene
8. Open the client from this PR again
9. It should connect to 100,100
10. Teleport to a world, for example `/goto olavra` (Same issue, it wont load, but doesnt matter)
11. Check the world loads correctly
12. Press the button "Go back to Genesis City"
13. It should take you to 100,100 (During the festival the avatar should spawn in the festival coordinates)

### CODE REVIEW

**Review by fcolarich — APPROVED** [review]
Nice!! thank you for fixing my mess 😬

**Comment by AlejandroAlvarezMelucciDCL** [comment]
> Nice!! thank you for fixing my mess 😬

It's _our_ mess
image

File: Explorer/Assets/DCL/UserInAppInitializationFlow/StartupOperations/SetOverrideStartingParcelOperation.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+using DCL.FeatureFlags;
+using ECS.SceneLifeCycle.Realm;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+using Utility.Types;
+
+namespace DCL.UserInAppInitializationFlow.StartupOperations
+{
+    public class SetOverrideStartingParcelOperation : IStartupOperation
```
Comment (dalkia) [inline Explorer/Assets/DCL/UserInAppInitializationFlow/StartupOperations/SetOverrideStartingParcelOperation.cs:12]: I think this logic can be alltogether in `TeleportStartupOperation`. If you do that, you have the power to modify `startParcel` straight away, therefore the teleport would work just fine.
And `RealmNavigator` will know nothing of its overridden values, so you can safely remove al the extra logic added there.

File: Explorer/Assets/DCL/UserInAppInitializationFlow/StartupOperations/SetOverrideStartingParcelOperation.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            this.featureFlagsCache = featureFlagsCache;
+            this.isLocalDevelopment = isLocalDevelopment;
+        }
+
+        public async UniTask<Result> ExecuteAsync(AsyncLoadProcessReport report, CancellationToken ct)
+        {
+            string? parcelToTeleportOverride = null;
+
+            bool overrideStartingParcel = !isLocalDevelopment &&
+                                          startParcel == Vector2Int.zero &&
```
Comment (dalkia) [inline Explorer/Assets/DCL/UserInAppInitializationFlow/StartupOperations/SetOverrideStartingParcelOperation.cs:36]: @fcolarich will this work if I want to go to 0,0; but the start parcel is elsewhere? 

Wouldnt it just take me to the elsewhere place?

File: Explorer/Assets/DCL/UserInAppInitializationFlow/StartupOperations/SetOverrideStartingParcelOperation.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                                          featureFlagsCache.Configuration.IsEnabled(FeatureFlagsStrings.GENESIS_STARTING_PARCEL) &&
+                                          featureFlagsCache.Configuration.TryGetTextPayload(FeatureFlagsStrings.GENESIS_STARTING_PARCEL, FeatureFlagsStrings.STRING_VARIANT, out parcelToTeleportOverride) &&
+                                          parcelToTeleportOverride != null;
+
+            realmNavigator.SetOverrideValues(overrideStartingParcel, parcelToTeleportOverride!);
+
+            await Task.CompletedTask;
```
Comment (dalkia) [inline Explorer/Assets/DCL/UserInAppInitializationFlow/StartupOperations/SetOverrideStartingParcelOperation.cs:43]: This line is not required, can be deleted

**Review by dalkia — APPROVED** [review]
Its looking way better! Nice

I do have some comments though. I wont delay the PR due to its criticality, but please consider the comments when you have some time.

**Review by DafGreco — APPROVED** [review]
Verified on both platforms Windows and Macos ! ✅

File: Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs [hunk 4]
```diff
@@ -44,22 +44,6 @@ public void DoNotSetDevelopmentModeIfMissingLocalSceneParam()
             Assert.AreEqual(70, realmLaunchSettings.TargetScene.y);
         }
 
-        [Test]
-        public void ApplyStartingPositionFromAppArgs()
```
Comment (AlejandroAlvarezMelucciDCL) [inline Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs:48]: Maybe we should move this test elsewhere instead of deleting it completely?

**Review by AlejandroAlvarezMelucciDCL — COMMENTED** [review]
LGTM, just left a small question @dalkia 👍 
And also let's track if more work is needed for the edge case we discussed when parcels aren't (or are, can't remember which one caused the issue) at 0,0

File: Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs (hunk 4 above)
Comment (dalkia) [reply Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs:48]: That test is checking the initial position from the RealmLaunchSettings. It have been removed entirely from that domain

Comment (AlejandroAlvarezMelucciDCL) [reply Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs:48]: I know, but do we still have that launch param tested elsewhere?

File: Explorer/Assets/DCL/UserInAppInitializationFlow/StartupOperations/TeleportStartupOperation.cs [hunk 5]
```diff
@@ -46,14 +49,18 @@ private void CheckParcelOverride()
...
-            //First we need to check if the user has passed a position as an argument. This is the case used on local scene development
+            //First we need to check if the user has passed a position as an argument. This is the case used on local scene development from creator hub/scene args
             //Check https://github.com/decentraland/js-sdk-toolchain/blob/2c002ca9e6feb98a771337190db2945e013d7b93/packages/%40dcl/sdk-commands/src/commands/start/explorer-alpha.ts#L29
             if (appArgs.TryGetValue(AppArgsFlags.POSITION, out parcelToTeleportOverride))
             {
                 ParsePositionAppParameter(parcelToTeleportOverride);
                 return;
             }
 
+            //If we are in editor, and its LSD, we want to ignore the feature falg
```
Comment (AlejandroAlvarezMelucciDCL) [inline Explorer/Assets/DCL/UserInAppInitializationFlow/StartupOperations/TeleportStartupOperation.cs:60]: Typo in "falg"

**Review by lorux0 — CHANGES_REQUESTED** [review]

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/Realm/IRealmNavigator.cs [hunk 6]
```diff
@@ -20,6 +20,9 @@ public interface IRealmNavigator
         public const string SDK_TEST_SCENES_URL = "https://sdk-team-cdn.decentraland.org/ipfs/sdk7-test-scenes-main-latest";
         public const string TEST_SCENES_URL = "https://sdk-test-scenes.decentraland.zone";
 
+        // True if changed to GenesisCity, False - when changed to any other realm
+        event Action<bool> RealmChanged;
```
[the remark this answers is not in the document]
Comment (dalkia) [reply Explorer/Assets/Scripts/ECS/SceneLifeCycle/Realm/IRealmNavigator.cs:24]: Lets make that one on another PR if necessary. I don't see the need to do it in this one in particular, it would just add an unnecessary change

File: Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs (hunk 4 above)
Comment (dalkia) [reply Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs:48]: Brought back

**Review by lorux0 — APPROVED** [review]

File: Explorer/Assets/Scripts/Global/Dynamic/RealmLaunchSettings.cs [hunk 7]
```diff
@@ -156,5 +161,28 @@ private bool IsRealmAWorld(string realmParam) =>
         private bool IsRealmAValidUrl(string realmParam) =>
             Uri.TryCreate(realmParam, UriKind.Absolute, out Uri? uriResult)
             && (uriResult.Scheme == Uri.UriSchemeHttp || uriResult.Scheme == Uri.UriSchemeHttps);
+
+        public void CheckStartParcelFeatureFlagOverride(IAppArgs appArgs, FeatureFlagsCache featureFlagsCache)
```
Comment (lorux0) [inline Explorer/Assets/Scripts/Global/Dynamic/RealmLaunchSettings.cs:165]: Perhaps rename into `ApplyConfig` overload? As we could apply any other config from the feature flags (probably BDUF tho).

File: Explorer/Assets/Scripts/Global/Dynamic/Bootstraper.cs [hunk 8]
```diff
@@ -224,6 +220,11 @@ public async UniTask LoadStartingRealmAsync(DynamicWorldContainer dynamicWorldCo
             await dynamicWorldContainer.RealmController.SetRealmAsync(startingRealm.Value, ct);
         }
 
+        public void CheckInitialParcelFeatureFlagOverride(FeatureFlagsCache featureFlagsCache)
```
Comment (lorux0) [inline Explorer/Assets/Scripts/Global/Dynamic/Bootstraper.cs:223]: Consider also renaming this one into `ApplyFeatureFlagConfigs`

Comment (dalkia) [reply Explorer/Assets/Scripts/Global/Dynamic/Bootstraper.cs:223]: I like it renaming here. Changing

File: Explorer/Assets/Scripts/Global/Dynamic/RealmLaunchSettings.cs (hunk 7 above)
Comment (dalkia) [reply Explorer/Assets/Scripts/Global/Dynamic/RealmLaunchSettings.cs:165]: Not so fond of changing it here, since this method includes logic very start parcel related (check if the parcel has changed in the Args)

So maybe we can keep this one separated

**Comment by DafGreco** [comment]
Verified on both platforms Windows and Macos ! ✅

 ✅Start the client with the flags --debug --dclenv zone. You should start in 100,100
 ✅Start the client with the flags --debug --dclenv zone --position 150,150. You should start in 150,150
 ✅Start the client with the flags --debug --dclenv zone --realm olavra.dcl.eth.

**Comment by DafGreco** [comment]
All commands seem to work properly on both platforms ! ✅
